### PR TITLE
enhancement: add auth config to pull image in private registry

### PR DESCRIPTION
**Pull request by calmkart**
for this [issue](https://github.com/aylei/kubectl-debug/issues/53)
add auth config to pull image in private registry

there are two way to set auth config:
1. kubectl-debug --image imageName --username userName --password passWord
2. edit the  ~/.kube/debug-config , add
```
authConfig:
    username: calmkart
    password: somePassWordHere
```

i add this in readme.MD

@aylei please rebuild the debug_plugin,debug_agent and update the debug_agent image in dockerhub.

if there is any question, please let me know. Thanks

### CODE REVIEW

**Comment by calmkart** [comment]
> 
> 
> The idea LGTM, but IMHO, clear text password introduces security and management issues.
> 
> I'd recommend to reference the credential via a Kubernetes Secret to get rid of the credential management in our plugin.

ok, i'll change the auth way to k8s secret in few days.
do you think the username/password auth way is necessary? maybe we can Reserved it, and add a k8s secret auth way.

**Comment by aylei** [comment]
@calmkart I prefer not preserving the username/password auth, this is useful but maintaining an extra credential management module/system adds burdens and risks.

**Comment by calmkart** [comment]
@aylei i have change the auth way to k8s secret.
PTAL.
thanks!

**Comment by aylei** [comment]
@calmkart Sorry for the late response, I'm on vacation now and will take a look on the next Monday.

File: pkg/plugin/cmd.go [hunk 1]
```diff
@@ -442,11 +465,19 @@ func (o *DebugOptions) Run() error {
 		params := url.Values{}
 		params.Add("image", o.Image)
 		params.Add("container", containerID)
-		bytes, err := json.Marshal(o.Command)
+		var authStr string
+		registrySecret, err := o.CoreClient.Secrets(o.RegistrySecretNamespace).Get(o.RegistrySecretName, v1.GetOptions{})
+		if err != nil {
+			authStr = ""
```
Comment (aylei) [inline pkg/plugin/cmd.go]: We can safely skip auth if the secret is not found, but in other conditions, this error should be properly handled.

File: docs/zh-cn.md [hunk 2]
```diff
@@ -79,6 +79,11 @@ kubectl debug POD_NAME --port-forward --daemonset-ns=kube-system --daemonset-nam
 
 # 老版本的 kubectl 无法自动发现插件, 需要直接调用 binary
 kubectl-debug POD_NAME
+
+# 使用私有仓库镜像,并设置私有仓库使用的kubernetes secret
+# secret data原文请设置为 {Username: <username>, Password: <password>}
+# 默认secret_name为kubectl-debug-registry-secret,默认namspace为default
+kubectl-debug POD_NAME --image calmkart/netshoot:latest --registry-secret-name <k8s_secret_name> --registry-secret-namespace <namespace>
```
Comment (aylei) [inline docs/zh-cn.md]: I recommend to document that how to create this secret.

**Pull request merged**